includes/create-theme/theme-patterns.php

```diff
@@ -47,4 +71,111 @@ public static function create_pattern_link( $attributes ) {
+			file_put_contents( $file, $file_content );
+		}
+	}
+
+	public static function prepare_pattern_for_export( $pattern, $options = null ) {
+		if ( ! $options ) {
+			$options = array(
+				'localizeText'   => false,
+				'removeNavRefs'  => true,
+				'localizeImages' => true,
+			);
+		}
+
+		$pattern = CBT_Theme_Templates::eliminate_environment_specific_content( $pattern, $options );
+
+		if ( array_key_exists( 'localizeText', $options ) && $options['localizeText'] ) {
+			$pattern = CBT_Theme_Templates::escape_text_in_template( $pattern );
+		}
+
+		if ( array_key_exists( 'localizeImages', $options ) && $options['localizeImages'] ) {
+			$pattern = CBT_Theme_Media::make_template_images_local( $pattern );
+
+			// Write the media assets if there are any
+			if ( $pattern->media ) {
+				CBT_Theme_Media::add_media_to_local( $pattern->media );
+			}
+		}
+
+		return $pattern;
+	}
+
+	/**
+	 * Copy the local patterns as well as any media to the theme filesystem.
+	 */
+	public static function add_patterns_to_theme( $options = null ) {
+		$base_dir     = get_stylesheet_directory();
+		$patterns_dir = $base_dir . DIRECTORY_SEPARATOR . 'patterns';
+
+		$pattern_query = new WP_Query(
+			array(
+				'post_type'      => 'wp_block',
+				'posts_per_page' => -1,
+			)
+		);
+
+		if ( $pattern_query->have_posts() ) {
+			// If there is no patterns folder, create it.
+			if ( ! is_dir( $patterns_dir ) ) {
+				wp_mkdir_p( $patterns_dir );
+			}
+
+			foreach ( $pattern_query->posts as $pattern ) {
+				$pattern        = self::pattern_from_wp_block( $pattern );
+				$pattern        = self::prepare_pattern_for_export( $pattern, $options );
+				$pattern_exists = false;
+
+				// Check pattern is synced before adding to theme.
+				if ( 'unsynced' !== $pattern->sync_status ) {
+					// Check pattern name doesn't already exist before creating the file.
+					$existing_patterns = glob( $patterns_dir . DIRECTORY_SEPARATOR . '*.php' );
+					foreach ( $existing_patterns as $existing_pattern ) {
+						if ( strpos( $existing_pattern, 'pattern-' . $pattern->name . '.php' ) !== false ) {
+							$pattern_exists = true;
+						}
+					}
+
+					if ( $pattern_exists ) {
+						return new WP_Error(
+							'pattern_already_exists',
+							sprintf(
+								/* Translators: Pattern name. */
+								__(
+									'A pattern with this name already exists: "%s".',
+									'create-block-theme'
+								),
+								$pattern->name
+							)
+						);
+					}
+
+					// Create the pattern file.
+					$pattern_file = $patterns_dir . 'pattern-' . $pattern->name . '.php';
+					file_put_contents(
+						$patterns_dir . DIRECTORY_SEPARATOR . 'pattern-' . $pattern->name . '.php',
+						$pattern->content
+					);
+
+					self::replace_local_pattern_references( $pattern );
+
+					// Remove it from the database to ensure that these patterns are loaded from the theme.
+					wp_delete_post( $pattern->id, true );
```

Comment:
If I create a new pattern and save it using CBT, it reloads the editor but the post has been deleted. With Gutenberg active I at least get a better error response: 

**Without Gutenberg**
https://github.com/user-attachments/assets/79a8cd6a-6f0a-4a5e-9339-228b3b8da0e3

**Gutenberg 19.0.0**
 | ![Screenshot 2024-08-14 at 2 49 49 PM](https://github.com/user-attachments/assets/37537f1d-41f2-431e-882b-9b52493bb0cb)

Reply:
I received same error with latest Gutenberg trunk. When I disabled Gutenberg I got WSOD with HTML `<title>null ‹ Template ‹ Gutenberg Trunk ‹ Editor — WordPress</title>` 🤔

Reply:
> I wonder if we can handle this redirection better, maybe return the user back to the patterns view?

Yes, nice idea, I've done this here: https://github.com/WordPress/create-block-theme/pull/675/commits/6de813f38bcce5951cf35ff6ac33d36cda4a6318.

Thanks both!